homeassistant/components/smarttub/binary_sensor.py

```diff
@@ -38,3 +53,45 @@ def is_on(self) -> bool:
     def device_class(self) -> str:
         """Return the device class for this entity."""
         return DEVICE_CLASS_CONNECTIVITY
+
+
+class SmartTubReminder(SmartTubEntity, BinarySensorEntity):
+    """Reminders for maintenance actions."""
+
+    def __init__(self, coordinator, spa, reminder):
+        """Initialize the entity."""
+        super().__init__(
+            coordinator,
+            spa,
+            f"{reminder.name.title()} Reminder",
+        )
+        self.reminder_id = reminder.id
+
+    @property
+    def unique_id(self):
+        """Return a unique id for this sensor."""
+        return f"{self.spa.id}-reminder-{self.reminder_id}"
+
+    @property
+    def reminder(self) -> SpaReminder:
+        """Return the underlying SpaReminder object for this entity."""
+        return self.coordinator.data[self.spa.id]["reminders"][self.reminder_id]
+
+    @property
+    def is_on(self) -> bool:
+        """Return whether the specified maintenance action needs to be taken."""
+        return self.reminder.remaining_days == 0
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        when = datetime.now() + timedelta(days=self.reminder.remaining_days)
```

Comment:
This is a bit tricky, because the reminder is based on the device clock which is not time zone aware (mine is currently 1 hour behind local time because we recently began DST).

I think I should just remove this attribute, until I can implement it more correctly.